### ISSUE TITLE
Rule to prevent redundant `Promise.resolve` or `Promise.reject`

### ISSUE THREAD

**Issue by ForbesLindesay** [eslint/eslint#5760]
**When does this rule warn?**

When wrapping a value returned from a Promise handlerFn in `Promise.resolve` or `Promise.reject`

```js
something.then(function () {
  // `Promise.resolve` is redundant, just use `return val;`
  return Promise.resolve(val);
});

// and

something.then(function () {
  // `Promise.reject` is redundant, just use `throw val;`
  return Promise.reject(val);
});
```

**Is this rule preventing an error or is it stylistic?**

It helps users to understand how promise values propagate.

**Why is this rule a candidate for inclusion instead of creating a custom rule?**

Promises are widely used, but currently still not well understood.  Including this rule in core will help people learn how to correctly use promises.

**Are you willing to create the rule yourself?**

Yes

**Comment by mysticatea** [comment]
Thank you for this issue.

I think this is stylistic rule since this rule doesn't catch any error.

Though we cannot confirm whether that `something` is surely an object of ES2015 Promise or not, there are many Promise-like libraries (e.g. jQuery.Deffered, Q, Bluebird, ...), do those have the same spec?

**Comment by ilyavolodin** [comment]
I think this should be added to [eslint-plugin-promise](https://github.com/xjamundx/eslint-plugin-promise) instead of the core.

**Comment by alberto** [comment]
Agreed
